client/extensions/zoninator/state/data-layer/zones/test/index.js

```diff
@@ -71,3 +94,132 @@ describe( '#requestZonesError', () => {
 	} );
 } );
+
+describe( '#createZone()', () => {
+	it( 'should dispatch a HTTP request to create a new zone', () => {
+		const dispatch = sinon.spy();
+		const action = {
+			type: 'DUMMY_ACTION',
+			siteId: 123456,
+			data: zone,
+			form: 'form',
+		};
+
+		createZone( { dispatch }, action );
+
+		expect( dispatch ).to.have.been.calledWith( http( {
+			method: 'POST',
+			path: '/jetpack-blogs/123456/rest-api/',
+			query: {
+				body: JSON.stringify( zone ),
+				json: true,
+				path: '/zoninator/v1/zones',
+			},
+		}, action ) );
+	} );
+
+	it( 'should dispatch `startSave`', () => {
+		const dispatch = sinon.spy();
+		const action = {
+			type: 'DUMMY_ACTION',
+			siteId: 123456,
+			data: zone,
+			form: 'form',
+		};
+
+		createZone( { dispatch }, action );
+
+		expect( dispatch ).to.have.been.calledWith( startSave( 'form' ) );
+	} );
+
+	it( 'should dispatch `removeNotice`', () => {
+		const dispatch = sinon.spy();
+		const action = {
+			type: 'DUMMY_ACTION',
+			siteId: 123456,
+			data: zone,
+			form: 'form',
+		};
+
+		createZone( { dispatch }, action );
+
+		expect( dispatch ).to.have.been.calledWith( removeNotice( 'zoninator-zone-create' ) );
+	} );
+} );
+
+describe( '#announceZoneSaved()', () => {
+	it( 'should dispatch `stopSave`', () => {
+		const dispatch = sinon.spy();
+		const action = {
+			type: 'DUMMY_ACTION',
+			siteId: 123456,
+			data: zone,
+			form: 'form',
+		};
+
+		announceZoneSaved( { dispatch }, action, null, { data: zone } );
+
+		expect( dispatch ).to.have.been.calledWith( stopSave( 'form' ) );
+	} );
+
+	it( 'should dispatch `updateZone`', () => {
+		const dispatch = sinon.spy();
+		const action = {
+			type: 'DUMMY_ACTION',
+			siteId: 123456,
+			form: 'form',
+		};
+
+		announceZoneSaved( { dispatch }, action, null, { data: zone } );
+
+		expect( dispatch ).to.have.been.calledWith( updateZone( 123456, zone ) );
+	} );
+
+	it( 'should dispatch `successNotice`', () => {
+		const dispatch = sinon.spy();
+		const action = {
+			type: 'DUMMY_ACTION',
+			siteId: 123456,
+			form: 'form',
+		};
+
+		announceZoneSaved( { dispatch }, action, null, { data: zone } );
+
+		expect( dispatch ).to.have.been.calledWith( successNotice(
+			translate( 'Zone saved!' ),
+			{ id: 'zoninator-zone-create' },
+		) );
+	} );
+} );
+
+describe( '#announceFailure()', () => {
+	it( 'should dispatch `stopSave`', () => {
+		const dispatch = sinon.spy();
+		const action = {
+			type: 'DUMMY_ACTION',
+			siteId: 123456,
+			form: 'form',
+		};
+
+		announceFailure( { dispatch }, action );
+
+		expect( dispatch ).to.have.been.calledWith( stopSave( 'form' ) );
+	} );
+
+	it( 'should dispatch `errorNotice`', () => {
+		const dispatch = sinon.spy();
+		const action = {
+			type: 'DUMMY_ACTION',
+			siteId: 123456,
+			data: zone,
+			form: 'form',
+		};
+
+		announceFailure( { dispatch }, action );
+
+		expect( dispatch ).to.have.been.calledWith( errorNotice(
+			translate( 'There was a problem saving the zone. Please try again.' ),
```

Comment:
Hi! I've found a possible matching string that has already been translated **20** times:
`translate( 'There was a problem saving your changes. Please try again.' )` *ES Score: 10*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/zoninator-new-zone-data-layer)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).